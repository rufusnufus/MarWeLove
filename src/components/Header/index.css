.header {
  background: #202020;
  height: 74px;
  display: flex;
}

.header__logo {
  background: #ec1d24;
  color: black;
  width: 271px;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  font-size: 50px;
  line-height: 74px;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
  margin-left: 60px;
}

.header__logo > span {
  color: white;
  font-size: 37px;
}

.header__navbar {
  font-size: 22px;
  line-height: 74px;
  margin-left: 60px;
}

.header__button {
  color: #ffffff;
  margin-right: 60px;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
}
