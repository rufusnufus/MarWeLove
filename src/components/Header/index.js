import './index.css';

function Header() {
  return (
    <div className="header">
        <a href="#" className="header__logo">
            Mar<span>we</span>Love
        </a>
        <div className="header__navbar">
            <a href="#" className="header__button">
                characters
            </a>
            <a href="#" className="header__button">
                comics
            </a>
        </div>
    </div>
  );
}

export default Header;