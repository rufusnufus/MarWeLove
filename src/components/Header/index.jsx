import React from 'react'
import { NavLink } from 'react-router-dom'

import './index.css'

function Header({ name }) {
  return (
    <header className="header">
      <NavLink data-testid="header__logo" className="header__logo" to="/">
        Mar<span>we</span>Love
      </NavLink>
      <div className="header__navbar">
        <NavLink className="header__button" to="/characters">
          characters
        </NavLink>
        <NavLink className="header__button" to="/comics">
          comics
        </NavLink>
        <NavLink className="header__button" to="/bookmarks">
          bookmarks
        </NavLink>
      </div>
      <span className="header__name">{name}</span>
    </header>
  )
}

export default Header
