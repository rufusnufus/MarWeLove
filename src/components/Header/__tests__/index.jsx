/**
 * @jest-environment jsdom
 */

import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { cleanup, render } from '@testing-library/react'
import Header from '../index'

afterEach(cleanup)

const mockHistoryReplace = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    replace: mockHistoryReplace
  })
}))

describe('Header', () => {
  it('Logo', () => {
    const { queryByTestId } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )

    expect(queryByTestId('header__logo')).toBeTruthy()
  })

  it('Characters', () => {
    const { queryByText } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )

    expect(queryByText('characters')).toBeTruthy()
  })

  it('Comics', () => {
    const { queryByText } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )

    expect(queryByText('comics')).toBeTruthy()
  })
})
