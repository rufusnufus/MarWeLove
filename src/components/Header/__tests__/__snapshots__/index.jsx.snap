// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render with name 1`] = `
<div>
  <header
    class="header"
  >
    <a
      aria-current="page"
      class="header__logo active"
      data-testid="header__logo"
      href="/"
    >
      Mar
      <span>
        we
      </span>
      Love
    </a>
    <div
      class="header__navbar"
    >
      <a
        class="header__button"
        href="/characters"
      >
        characters
      </a>
      <a
        class="header__button"
        href="/comics"
      >
        comics
      </a>
      <a
        class="header__button"
        href="/bookmarks"
      >
        bookmarks
      </a>
      <a
        class="header__button"
        href="/bookmarks_comics"
      >
        comic bookmarks
      </a>
    </div>
    <span
      class="header__name"
    >
      user
    </span>
  </header>
</div>
`;

exports[`Header Render without name 1`] = `
<div>
  <header
    class="header"
  >
    <a
      aria-current="page"
      class="header__logo active"
      data-testid="header__logo"
      href="/"
    >
      Mar
      <span>
        we
      </span>
      Love
    </a>
    <div
      class="header__navbar"
    >
      <a
        class="header__button"
        href="/characters"
      >
        characters
      </a>
      <a
        class="header__button"
        href="/comics"
      >
        comics
      </a>
      <a
        class="header__button"
        href="/bookmarks"
      >
        bookmarks
      </a>
      <a
        class="header__button"
        href="/bookmarks_comics"
      >
        comic bookmarks
      </a>
    </div>
    <span
      class="header__name"
    />
  </header>
</div>
`;
